### PR TITLE
StorageService resurrection

### PR DESCRIPTION
**Pull request by jasonwyatt**
This PR adds the ability for the StorageService to use its superclass: ResurrectorService to resurrect clients when one of those clients updates the store.

### CODE REVIEW

File: java/arcs/android/storage/service/BindingContextStats.kt [hunk 1]
```diff
@@ -21,7 +32,8 @@ interface BindingContextStatisticsSink {
  *
  * **Note:** This makes the assumption that the distribution of processing times is normal.
  */
-interface BindingContextStatistics : BindingContextStatisticsSink {
+interface BindingContextStatistics :
+    BindingContextStatisticsSink {
```
Comment (csilvestrini) [inline java/arcs/android/storage/service/BindingContextStats.kt]: Same line?

File: java/arcs/android/storage/service/BindingContextStats.kt [hunk 2]
```diff
@@ -41,7 +53,8 @@ interface BindingContextStatistics : BindingContextStatisticsSink {
 }
 
 @VisibleForTesting(otherwise = VisibleForTesting.PACKAGE_PRIVATE)
-class BindingContextStatsImpl : BindingContextStatistics {
+class BindingContextStatsImpl :
```
Comment (csilvestrini) [inline java/arcs/android/storage/service/BindingContextStats.kt]: Here too?

File: java/arcs/sdk/android/storage/ServiceStore.kt [hunk 3]
```diff
@@ -123,7 +124,8 @@ class ServiceStore<Data : CrdtData, Op : CrdtOperation, ConsumerData>(
 
     override suspend fun onProxyMessage(message: ProxyMessage<Data, Op, ConsumerData>): Boolean {
         val service = checkNotNull(storageService)
-        val result = DeferredResult(coroutineContext)
+        val result =
```
Comment (csilvestrini) [inline java/arcs/sdk/android/storage/ServiceStore.kt]: Same line

File: javatests/arcs/android/storage/service/ParcelableProxyMessageChannelTest.kt [hunk 4]
```diff
@@ -72,7 +76,8 @@ class ParcelableProxyMessageChannelTest {
         try {
             supervisorScope {
                 val channel = ParcelableProxyMessageChannel(coroutineContext)
-                val deferredResult = DeferredResult(coroutineContext)
+                val deferredResult =
```
Comment (csilvestrini) [inline javatests/arcs/android/storage/service/ParcelableProxyMessageChannelTest.kt]: Same line

File: javatests/arcs/sdk/android/storage/ServiceStoreTest.kt [hunk 5]
```diff
@@ -80,7 +80,8 @@ class ServiceStoreTest {
         assertThat(data.values).isEmpty()
 
         // Increment at the binding context directly.
-        val deferredResult = DeferredResult(coroutineContext)
+        val deferredResult =
+            DeferredResult(coroutineContext)
```
Comment (csilvestrini) [inline javatests/arcs/sdk/android/storage/ServiceStoreTest.kt]: Same line

File: javatests/arcs/sdk/android/storage/service/StorageServiceTest.kt [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+                listOf(op), id = 1
+            )
+            val deferredResult = DeferredResult(this.coroutineContext)
+            context.sendProxyMessage(
+                proxyMessage.toParcelable(ParcelableCrdtType.Count),
+                deferredResult
+            )
+
+            deferredResult.await()
+        }.let(::assertThat).isTrue()
```
Comment (csilvestrini) [inline javatests/arcs/sdk/android/storage/service/StorageServiceTest.kt]: This line is super weird. I can guess at what it does, but surely it'd be better to just use a temporary variable for this?

```
val result = runBlocking { ... }
assertThat(result ?: true).isTrue()
```

File: java/arcs/android/storage/service/BindingContextStats.kt (hunk 1 above)
Comment (jasonwyatt) [reply java/arcs/android/storage/service/BindingContextStats.kt]: Ah crap, Refactor->Move loves to try to help out by reformatting, and it ends up wrapping too aggressively.

File: javatests/arcs/sdk/android/storage/service/StorageServiceTest.kt (hunk 6 above)
Comment (jasonwyatt) [reply javatests/arcs/sdk/android/storage/service/StorageServiceTest.kt]: I was being too clever, yeah.. 🙃

**Comment by jasonwyatt** [comment]
I've verified that the StorageServiceTest works well on my workstation, laptop, and even internally. For some reason Travis just isn't having it though... so I've @Ignore'd it for now.  Maybe if we switch to Cloud Build or Circle, it'd be worth trying again.

**Comment by jasonwyatt** [comment]
Locally merged in cl/289763763

**Pull request merged**